### PR TITLE
Workflow doc

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

File: en/user_guide/workflow.rst [hunk 1]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
 It allows several levels of validation by different people from creation to publication. Workflow
 are fully customizable.
 
-A workflow is composed of steps, each corresponding to a status. Each status is linked to one (or
-several) other status by a transition. To change a content status or a node status, a contributor
-must have a right mapping the corresponding transition.
+A workflow is composed of steps, each corresponding to a status.
+Each status could be linked to an other by a transition named 'Role'.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: Not sure but maybe called rather than named

File: en/user_guide/workflow.rst [hunk 2]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+These roles are grouped in collection named 'Workflow Function'.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: same here

Comment (ngilain) [inline en/user_guide/workflow.rst]: in a collection or in collections

File: en/user_guide/workflow.rst [hunk 3]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+Finally, in user's administration, it is possible to add a workflow function to a user for a given content type or for nodes.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: for a given content or on a given content ?

File: en/user_guide/workflow.rst [hunk 4]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
 
-For instance, the article content type could have a workflow represented by the following status:
+For example :
+    - three statuses : draft, pending and published
+    - two roles : DRAFT_TO_PUBLISHED (transition draft to published), PENDING_TO_PUBLISHED (transition pending to published)
+    - one workflow function: Validator (containing DRAFT_TO_PUBLISHED and PENDING_TO_PUBLISHED)
+    - a user with Validator function on News (content type)
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: You should use `` in this paragraph to distinguish the names of your example: `draft`, `DRAFT_TO_PUBLISHED`, `validator` and so on

File: en/user_guide/workflow.rst [hunk 5]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
 
-    Draft => Pending => Published
+There are two kinds of roles, those with a from and a to status and those without these statuses.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: `` around from and to status

Comment (ngilain) [inline en/user_guide/workflow.rst]: without this status links

File: en/user_guide/workflow.rst [hunk 6]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+First one are workflow roles and the only ones which could be used in workflow function.
+Second one are access roles and are used to grant access or not in BackOffice. They are used in group context.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: to grant access or not to some Back Office features.

File: en/user_guide/workflow.rst [hunk 7]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
 * **Name** (required): status name
 * **Published**: if this status is final
 * **Initial**: if this status is initial
-* **Label**: status label in all available languages
+* **Label**: status label in all available languages in BackOffice
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: in the Back Office

File: en/user_guide/workflow.rst [hunk 8]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
...
-transition an administrator only have to create a new right. In Open Orchestra, rights are called
-Roles.
+Roles
+-----
 
 The role form asks for the following information:
 
 * **Name**: role name
-* **Description**: role description in all available languages.
+* **Description**: role description in all available languages in BackOffice.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: in the Back Office

File: en/user_guide/workflow.rst [hunk 9]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
...
+* **Status from**: as roles are transitions between two statuses, source status for the role
+* **Status destination**: as roles are transitions between two statuses, destination status for the role
 
 .. image:: ../../images/role_form.png
 
-The selection of two status in the role form will define a transition role, and implicitly this transition.
+The selection of two status in the role form will define a workflow role.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: status or statuses ?

File: en/user_guide/workflow.rst [hunk 10]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
...
 
 This role creates a transition between Draft status and Pending status. So users with this role will
 be able to change the status of a node or content from Draft to Pending. They will do this via a small
-widget on the node edit page.
+widget on the node or content edit page.
 
 .. image:: ../../images/status_transition.png
 
-See also `group management`_ to know how to give roles to the users.
+If no statuses are selected, the role will define an access role. See also `group management`_
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: If no status is selected

File: en/user_guide/workflow.rst [hunk 11]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
...
+
+Workflow Function
+-----------------
+
+The workflow function form asks for the following information:
+
+* **Name**: workflow function name in all available languages in BackOffice.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: in the Back Office

File: en/user_guide/workflow.rst [hunk 12]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
...
+* **Role**: a list of all roles with a from and destination status, multiple selection allowed.
```
Comment (ngilain) [inline en/user_guide/workflow.rst:83]: Maybe we should call once for all the two types of roles workflow roles and acces roles. That way we could avoid this kind of heavy description 'a role with/without a from and destination status'
That naming could be done earlier when you introduce that two kinds of role

File: en/user_guide/workflow.rst (hunk 10 above)
Comment (veolys-nanne) [inline en/user_guide/workflow.rst]: Both are correct for me. I keep plural. @itkg-ngilain

**Comment by veolys-nanne** [comment]
done @itkg-ngilain

File: en/user_guide/workflow.rst [hunk 13]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
-several) other status by a transition. To change a content status or a node status, a contributor
-must have a right mapping the corresponding transition.
+A workflow is composed of steps, each corresponding to a status.
+Each status could be linked to an other by a transition called 'Role'.
+These roles are grouped in collections called 'Workflow Function'.
+Finally, in user's administration, it is possible to add a workflow function to a user on a given content type or for nodes.
 
-For instance, the article content type could have a workflow represented by the following status:
+For example :
+    - three statuses : "draft", "pending" and "published"
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: It's not the good character ^^ you must use ``: it's a rst syntax interpreted by github making the text underlined

File: en/user_guide/workflow.rst [hunk 14]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
-A workflow is composed of steps, each corresponding to a status. Each status is linked to one (or
-several) other status by a transition. To change a content status or a node status, a contributor
-must have a right mapping the corresponding transition.
+A workflow is composed of steps, each corresponding to a status.
+Each status could be linked to an other by a transition called 'Role'.
+These roles are grouped in collections called 'Workflow Function'.
+Finally, in user's administration, it is possible to add a workflow function to a user on a given content type or for nodes.
 
-For instance, the article content type could have a workflow represented by the following status:
+For example :
```
Comment (nicolasThal) [inline en/user_guide/workflow.rst]: For instance ?

File: en/user_guide/workflow.rst [hunk 15]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+    - three statuses : "draft", "pending" and "published"
+    - two roles : "DRAFT_TO_PUBLISHED" (transition "draft" to "published"), "PENDING_TO_PUBLISHED" (transition "pending" to "published")
```
Comment (nicolasThal) [inline en/user_guide/workflow.rst]: The first status should not be the transition from draft to pending ?

File: en/user_guide/workflow.rst [hunk 16]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+    - one workflow function: "Validator" (containing "DRAFT_TO_PUBLISHED" and "PENDING_TO_PUBLISHED")
+    - a user with "Validator" function on "News" (content type)
 
-    Draft => Pending => Published
+There are two kinds of roles, those with link to a "from" and a "to status" and those without this status links.
```
Comment (nicolasThal) [inline en/user_guide/workflow.rst]: those creating a link between a _from status_ and a _to status_, and those without this status link.

File: en/user_guide/workflow.rst [hunk 17]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+First one are workflow roles and the only ones which could be used in workflow function.
```
Comment (nicolasThal) [inline en/user_guide/workflow.rst]: The first type are the workflow roles, the only type that could be used in workflow function

File: en/user_guide/workflow.rst [hunk 18]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
+Second one are access roles and are used to grant access or not to Back Office features. They are used in group context.
```
Comment (nicolasThal) [inline en/user_guide/workflow.rst]: The second one are access roles, used to grant access to the different part of the Back Office.

File: en/user_guide/workflow.rst (hunk 15 above)
Comment (veolys-nanne) [inline en/user_guide/workflow.rst]: This is an example. As I want to show it with an validator function, i choose any status to published. @nicolasThal

**Comment by veolys-nanne** [comment]
done @nicolasThal , @itkg-ngilain

File: en/user_guide/workflow.rst [hunk 19]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
 It allows several levels of validation by different people from creation to publication. Workflow
 are fully customizable.
 
-A workflow is composed of steps, each corresponding to a status. Each status is linked to one (or
-several) other status by a transition. To change a content status or a node status, a contributor
-must have a right mapping the corresponding transition.
+A workflow is composed of steps, each corresponding to a status.
+Each status could be linked to an other by a transition called 'Role'.
+These roles are grouped in collections called 'Workflow Function'.
+Finally, in user's administration, it is possible to add a workflow function to a user on a given content type or for nodes.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: or on nodes

File: en/user_guide/workflow.rst [hunk 20]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
 
-For instance, the article content type could have a workflow represented by the following status:
+For example :
+    - three statuses : ``draft``, ``pending`` and ``published``
+    - two roles : ``DRAFT_TO_PUBLISHED`` (transition ``draft`` to ``published``), ``PENDING_TO_PUBLISHED`` (transition ``pending`` to ``published``)
+    - one workflow function: ``Validator`` (containing ``DRAFT_TO_PUBLISHED`` and ``PENDING_TO_PUBLISHED``)
+    - a user with ``Validator`` function on ``News`` (content type)
 
-    Draft => Pending => Published
+There are two kinds of roles, those creating a link between a ``from`` and a ``to status`` and those without this status links.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: this statuses link

File: en/user_guide/workflow.rst [hunk 21]
```diff
@@ -5,21 +5,23 @@ In Open Orchestra, key contributions such as nodes and contents are subject to a
...
 
-In that configuration, three different types of contributors could be involved in the validation of an
-article. A contributor create the content and make it in the Draft status. A second one is responsible
-for a first step of validation and, if ok, put it in the Pending status. A last contributor is required
-to publish the content on the Front Office.
-
-So to create a workflow, an administrator needs to create the different status, then the transitions
-linking those status, and finally assign rights to the contributors.
+When adding a workflow function to a user on a particular content type or on the nodes, it is possible to choose
+if this function is activated only for the contributions of this user or for all contributions by checking owned.
```
Comment (ngilain) [inline en/user_guide/workflow.rst]: by checking ``owned``

File: en/user_guide/workflow.rst [hunk 22]
```diff
@@ -43,39 +45,51 @@ The status form asks for the following information:
...
+
+Workflow Function
+-----------------
+
+The workflow function form asks for the following information:
+
+* **Name**: workflow function name in all available languages in Back Office.
+* **Role**: a list of all roles with a from and destination status, multiple selection allowed.
+
+.. image:: ../../images/workflow_function_form.png
```
Comment (ngilain) [inline en/user_guide/workflow.rst:85]: Roles have been recently translated, so I suggest to update that screen to make constants disappear

**Comment by veolys-nanne** [comment]
done @itkg-ngilain, @nicolasThal

**Comment by ngilain** [comment]
:+1:

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**